Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -177,6 +176,37 @@ public AsyncFunctionAssertions(Func<TTask> subject, IExtractExceptions extractor
         return new ExceptionAssertions<TException>(Array.Empty<TException>());
     }
 
+    /// <summary>
+    /// Asserts that the current <see cref="Func{Task}"/> throws an exception of type <typeparamref name="TException"/>.
+    /// </summary>
+    /// <typeparam name="TException">The type of exception expected to be thrown.</typeparam>
+    /// <param name="timeSpan">The allowed time span for the operation.</param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public async Task<ExceptionAssertions<TException>> ThrowWithinAsync<TException>(
+        TimeSpan timeSpan, string because = "", params object[] becauseArgs)
+        where TException : Exception
+    {
+        bool success = Execute.Assertion
+            .ForCondition(Subject is not null)
+            .BecauseOf(because, becauseArgs)
+            .FailWith("Expected {context} to throw {0} within {1}{reason}, but found <null>.",
+                typeof(TException), timeSpan);
+
+        if (success)
+        {
+            Exception exception = await InvokeWithInterceptionAsync(timeSpan);
+            return ThrowInternal<TException>(exception, timeSpan, because, becauseArgs);
```

Comment:
🔧 This method implies it's throwing an exception, but I think the point is to assert that a particular exception was thrown. Let's name it accordingly.

Reply:
It's `private protected`, so how would it be a breaking change?

Reply:
The new one is private, the old one is protected. The old one was invented and named long time ago.
Both have similar behavior, so both should have similar name, isn't it? 
This is why I recommend consistent renaming with V7.

Reply:
When you move it back to `AsyncFunctionAssertions`, the naming becomes a local concern (as-in within the same class), so it becomes less important the naming is concistent.